### DISCUSSION TITLE
categorization with timeline variable not working properly

### DISCUSSION THREAD

**Discussion by halukoguz** [jspsych/jsPsych#2536] [Q&A] [answered]
Hey,
I am currently working on an experiment where the participants need to categorize whether a word is grammatically fitting or not. For that, I used the categorization plugin with a timeline variable. When running the code however, it does not show the right answer according to what I inserted into the variable. For better understanding, my code: 
```
var jn_trial= {
    timeline: [
        {
            type: "html-keyboard-response",
            stimulus: '<p style="font-size: 60px;">+</p>',
            choices: "NO_KEYS",
            trial_duration: 500,
            data: {
                task: 'fixation'
        }},
        {
            type: "categorize-html",
            stimulus: jsPsych.timelineVariable('nomen'),
            choices: [ "j", "f"],
            prompt: "<p> Passt das Demonstrativpronomen zu dem Nomen (Substantiv)? <p>",
            key_answer: jsPsych.timelineVariable('correct'),
            data: {
                task:"response",
        }
        },
    ],
    timeline_variables: [
        {nomen: "<p>dieser Kind</p>", correct: "f"},
        {nomen: "<p>diese Frau</p>", correct: "j"},
        {nomen: "<p>dieses Mann</p>", correct: "f"},
    ],

        randomize_order: true
}

timeline.push(jn_trial);
```

**Comment by jodeleeuw** [comment]
Hi @halukoguz 

What does the code do? At first glance this looks correct to me.

**Reply by halukoguz** [reply]
The code is working. But the feedback is in every trial "incorrect", no matter which response is given.

Thank you in advance.

**Comment by halukoguz** [comment, marked as answer]
I found the solution: apparently I have to give the Javascript Keycode (https://keycode.info/) into the "correct" space in the timeline variable. It works now, thank you!

**Reply by jodeleeuw** [reply]
Glad you found a solution! In case anyone else is seeing this in the future, I believe that this only applies to jsPsych v6.2 and earlier. Anything after v6.3 should accept the character value instead of the keycode.
